test/quantization/test_quantize_script.py

```diff
@@ -751,7 +751,33 @@ def forward(self, x):
             else:
                 m = torch.jit.script(M())
             m = prepare_script(m.eval(), {'': default_qconfig})
-            assert len(attrs_with_prefix(m, '_observer_',)) == 3
+            assert len(attrs_with_prefix(m, '_observer_')) == 3
+
+    def test_insert_observers_for_if_observed_before(self):
+        """ test case for values observed before if nodes
+        """
+        class M(torch.nn.Module):
+            def __init__(self, cond):
+                super(M, self).__init__()
+                self.conv = torch.nn.Conv2d(3, 3, 3).float()
+                self.cond = cond
+
+            def forward(self, x):
+                x = self.conv(x)
+                # x is already observed
+                if self.cond:
+                    x = torch.flatten(x)
```

Comment:
If this operation where instead 
```python
if self.cond:
  x = self.conv1(x)
return x
```
you would need to insert an observer  after conv1. Is this tested?

Reply:
not yet, I can add the test

Reply:
this does not work since different branches of `if` not quantized the same way

Reply:
there is a conflict of rules at the moment if we want to support this, we can probably think about it later

Reply:
Please file an issue to track this issue down the road.